### ISSUE TITLE
Feature Request add Intro skipping to the plugin

### ISSUE THREAD

**Issue by Dnkhatri** [im85288/service.upnext#89]
Hi I would like to request a new feature of skipping intros just like we can do currently with the outros with a timer. Currently I am using this plugin for that and it works only for the first episode I play does not work for episodes that play with upnext. I think this plugins concept of having a default intro skip time and ability to set it per series is a good implementation. Also I think the same could be adopted and implemented for the outros instead of how  upnext currently does so instead of just a default we could also get the ability to set the timer longer or  shorter series by series basis. 
 
https://github.com/aenemapy/aenemapyrepo/tree/master/script.tvskipintro

**Comment by dagwieers** [comment]
@Dnkhatri I am not a big an of this feature request. This will only work if you know in advance when the intro starts and stops. A lot of TV shows have the intro after the episode already started. In some episodes after 15 or 20 minutes, so you cannot even expect this to be reusable for the next episode.

I think it is fine as a niche add-on (tvskipintro) for people that would like this functionality, but it doesn't fit Up Next (in name or in intent).

**Comment by Dnkhatri** [comment]
1. The intro starting in the middle is an American tv show thing I think does not happen in all the other language shows that I have watched.
2. The plugin I posted allows disabling the skip intro per show basis so after watching the first episode you could disable it and it won't show up again for the next episode of the show
3.  This plugin is for binge watching i.e watching a few episodes of a show in a row. And the intros and outros hamper it so I personally think it is relevant to the plugin. But I am not a coder so if you guys think it can't be done or worth it I understand.

**Comment by im85288** [comment]
I do like the idea and indeed in my older addon (next up) one contributor added a skip intro integration...the problem we had though is that the only way to use it properly was by manually entering the start and end times of the intro for each and every episode...this is just not feasible.

The only place where this worked without any issues is with the Netflix addon which has proper meta data supplied with the episodes, this is why you will see that the notification time to play the next episode comes up exactly when the credits start (as the netflix addon states to this one when to do the pop up)..we could have done something similar with skipping the intros but the developer thought it made more sense to keep that in the Netflix addon.

So to summarise, unless the addon you linked has figured out a way to state the start/end time of an intro accurately for all episodes being played it is just to random to integrate in this addon

**Comment by dagwieers** [comment]
I am going to close this feature request. It has been moved to the "Revisit" milestone, which means we can reopen it if there is good reason to or when requested. Feel free to add feedback to this issue.

**Comment by Dnkhatri** [comment]
@dagwieers 
In the end I found a system that works for me by making edl files that skips the time I require for the first episode of a show then copying it for all the other episodes of the show. 
    Sadly can't do the same for the credits though as the runtime of the episodes of the same show can vary a lot.
   I know you guys don't want to make the plugin too complicated but was wondering if an advance option to generate edl files could be added to upnext plugin some of my shows have 2.5-3 minutes credits but as the episodes length of the same show varies too much can't make an edl file that removes part of the credits but at the same time can't increase the notification time for upnext to show up too early as other shows have no or very short credits. So an option to generate an edl files for all the episodes of a season/show when we get the first upnext notification could give finer control over skipping credits compared to the current guessing by runtime of an episode. BTW if you guys know of any app that can generate edl files per show or season let me know as that would be helpful but as upnext is not aware of the edl cropping out most of the credits. The upnext notification probably still won't show unless upnext supports. I am not a programer so I am not sure how hard or easy this would be to add to the plugin but putting it out there hoping someone thinks it is and builds it.
P.S Kodi intro skipping is broken but commercial skipping word so I am using that in my edl
